src/schema/resolvers/signin.js

```diff
@@ -0,0 +1,31 @@
+
+const INVALID_CREDENTIALS_ERROR_MESSAGE = 'Invalid username and/or password'
+
+export default {
+  Mutation: {
+    async signin (parent, { data: { username, password: passwordFromClient } }, context) {
```

Comment:
This may be clearer if you destruct these parameters in the function's body. Destructing is not very clear in itself.